xLights/EffectsPanel.cpp

```diff
@@ -210,7 +234,8 @@ wxString EffectsPanel::GetRandomEffectString(int effidx)
 
     // get effect controls
     wxWindow *window = EffectChoicebook->GetPage(effidx);
-    s += GetRandomEffectStringFromWindow(window, prefix);
+    wxWindow *wPanel = GetWindowPanel(window);
```

Comment:
Here is why the there were never any random settings generated, this was passing the wrong window with all the settings.

For reference i found this commit looks like it broke the randomization. https://github.com/smeighan/xLights/commit/b2e87fbd9f0151dc5610bb6ffe70f018504701f8